data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/LogstashAttributesMappings.java

```diff
@@ -35,6 +36,17 @@
      */
     Map<String, Object> getAdditionalAttributes();
 
+    /**
+     * A list of attribute names in the Logstash configuration which can have nested syntax.
+     * properties.
+     * <p>
+     * This should not return null and should return empty if not defined.
+     *
+     * @return A Map
+     * @since 1.2
+     */
+    List<String> getNestedSyntaxAttributeNames();
```

Comment:
I think this approach has helped reduce complexity for the simpler scenarios such as `kv`.